worker/worker_test.go

```diff
@@ -31,101 +32,81 @@ import (
 	_ "github.com/coreos/clair/worker/detectors/namespace/osrelease"
 )
 
-func TestProcessWithDistUpgrade(t *testing.T) {
-	// TODO(Quentin-M): This should not be bound to a single database implementation.
-	datastore, err := pgsql.OpenForTest("ProcessWithDistUpgrade", false)
-	if err != nil {
-		t.Error(err)
-		return
+type mockDatastore struct {
+	database.MockDatastore
+	layers map[string]database.Layer
+}
+
+func newMockDatastore() *mockDatastore {
+	return &mockDatastore{
+		layers: make(map[string]database.Layer),
 	}
-	defer datastore.Close()
+}
 
+func TestProcessWithDistUpgrade(t *testing.T) {
 	_, f, _, _ := runtime.Caller(0)
-	path := path.Join(path.Dir(f)) + "/testdata/DistUpgrade/"
+	testDataPath := path.Join(path.Dir(f)) + "/testdata/DistUpgrade/"
```

Comment:
Let's lean towards `filepath.Join` over `path.Join` just for clarity's sake.
Maybe one day Clair will run on Windows 🎱